mldsa/sign.h

```diff
@@ -102,6 +102,8 @@ int crypto_sign_signature_internal(uint8_t *sig, size_t *siglen,
                                    const uint8_t rnd[MLDSA_RNDBYTES],
                                    const uint8_t *sk, int externalmu)
 __contract__(
+  requires(mlen <= MLD_MAX_BUFFER_SIZE)
+  requires(prelen <= MLD_MAX_BUFFER_SIZE)
   requires(memory_no_alias(sig, CRYPTO_BYTES))
   requires(memory_no_alias(siglen, sizeof(size_t)))
   requires(memory_no_alias(m, mlen))
```

Comment:
A question possibly unrelated to this PR... if crypto_sign_signature_internal() is an internal function, why is it declared here in sign.h, rather than being local and static in sign.c?

Reply:
See https://github.com/pq-code-package/mldsa-native/issues/354